packages/@aws-cdk/aws-apigateway/lib/stage.ts

```diff
@@ -262,8 +299,68 @@ export class Stage extends Resource {
         loggingLevel: options.loggingLevel,
         metricsEnabled: options.metricsEnabled,
         throttlingBurstLimit: options.throttlingBurstLimit,
-        throttlingRateLimit: options.throttlingRateLimit,
+        throttlingRateLimit: options.throttlingRateLimit
       };
     }
   }
 }
+
+/**
+ * A API Gateway custom access log destination.
+ */
+export interface IAccessLogDestination {
+  /**
+   * Binds destination.
+   */
+  bind(): AccessLogDestinationConfig
+}
+
+/**
+ * Options when binding a destination to a API Gateway
+ */
+export interface AccessLogDestinationConfig {
+  /**
+   * The Amazon Resource Name (ARN) of the destination resource
+   */
+  readonly destinationArn: string;
+}
+
+/**
+ * Use CloudWatch Logs as a custom access log destination for API Gateway.
+ */
+export class CloudWatchLogsDestination implements IAccessLogDestination {
+  constructor(private readonly logGroup: ILogGroup) {
+  }
+
+  /**
+   * Binds this destination to the CloudWatch Logs.
+   */
+  public bind(): AccessLogDestinationConfig {
+    return {
+      destinationArn: this.logGroup.logGroupArn
+    };
+  }
+}
+
+/**
+ * Use Kinesis Data Firehose as a custom access log destination for API Gateway.
+ */
+export class KinesisDataFirehoseDestination implements IAccessLogDestination {
+  constructor(private readonly deliveryStreamArn: string) {
```

Comment:
When I started fiddling with this PR locally, I really enjoyed the simplicity of `ILogGroup` for CloudWatchLogsDestination and how it interacted with cloudwatch logs.

I am unsure how the interactions with Firehose within the cdk. Is there something simple like ILogGroup over there? My preference after pulling this is to have a "ILogGroup-esque experience" for firehose as well. If we believe that is deliveryStreamArn, then this works. Have we done the legwork to see if there is something similar in that space offered by the cdk?

Reply:
Just checked, it's unfortunate but there is no L2 construct for Kinesis Firehose: https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk/aws-kinesisfirehose

Not sure how I feel about the direct access though, since this API is essentially permanent :\

Reply:
It's not the worst thing in the world:

```ts
    const fh = new firehose.CfnDeliveryStream(this, 'TestFirehose', {
      // ...
    });
    
    const restApi = new apigw.RestApi(this, 'MyApi', {
      deployOptions: {
        accessLogDestination: new KinesisDataFirehoseDestination(fh.attrArn),
        accessLogFormat: JSON.stringify({
          requestId: "$context.requestId",
          // ...
        })
      }
    });

```

I have no hard feelings on this now since it unblocks use cases, and is the only way to interact with firehose to date.

Reply:
Yes, Firehose has only L2😭.
I have included the pseudo code you wrote in the document.